pkg/apis/calico/validation/networkconfig_test.go

```diff
@@ -0,0 +1,129 @@
+			ConsistOf(PointTo(MatchFields(IgnoreExtras, Fields{"Field": Equal("config.ipip")})))),
+		Entry("should succeed with valid IPIP mode", &apiscalico.NetworkConfig{IPIP: ptr.To(apiscalico.Always)}, field.NewPath("config"),
+			BeEmpty()),
+		Entry("should succeed with valid IP autodetection method", &apiscalico.NetworkConfig{IPAutoDetectionMethod: ptr.To("interface=cali1234")}, field.NewPath("config"),
+			BeEmpty()),
+		Entry("should succeed with valid IP autodetection method with interface and regex", &apiscalico.NetworkConfig{IPAutoDetectionMethod: ptr.To("interface=^cali.*")}, field.NewPath("config"),
+			BeEmpty()),
+		Entry("should succeed with valid IP autodetection method with interface", &apiscalico.NetworkConfig{IPAutoDetectionMethod: ptr.To("interface=cali1234")}, field.NewPath("config"),
+			BeEmpty()),
+		Entry("should succeed with valid IP autodetection method with a valid cidr", &apiscalico.NetworkConfig{IPAutoDetectionMethod: ptr.To("cidr=192.168.0.0/16")}, field.NewPath("config"),
+			BeEmpty()),
+
+		Entry("should return error with empty IP autodetection method", &apiscalico.NetworkConfig{IPAutoDetectionMethod: ptr.To("")}, field.NewPath("config"),
+			ConsistOf(PointTo(MatchFields(IgnoreExtras, Fields{"Field": Equal("config.ipAutoDetectionMethod")})))),
+
+		Entry("should return error with invalid IP autodetection method format", &apiscalico.NetworkConfig{IPAutoDetectionMethod: ptr.To("invalid-format")}, field.NewPath("config"),
+			ConsistOf(PointTo(MatchFields(IgnoreExtras, Fields{"Field": Equal("config.ipAutoDetectionMethod")})))),
+		Entry("should return error with invalid IP autodetection method regex", &apiscalico.NetworkConfig{IPAutoDetectionMethod: ptr.To("interface=invalid[regex")}, field.NewPath("config"),
+			ConsistOf(PointTo(MatchFields(IgnoreExtras, Fields{"Field": Equal("config.ipAutoDetectionMethod")})))),
+
+		Entry("should succeed with valid IP autodetection method with skip-interface", &apiscalico.NetworkConfig{IPAutoDetectionMethod: ptr.To("skip-interface=cali1234")}, field.NewPath("config"),
+			BeEmpty()),
+		Entry("should return error with invalid IP autodetection method with skip-interface", &apiscalico.NetworkConfig{IPAutoDetectionMethod: ptr.To("skip-interface=invalid[regex")}, field.NewPath("config"),
+			ConsistOf(PointTo(MatchFields(IgnoreExtras, Fields{"Field": Equal("config.ipAutoDetectionMethod")})))),
+		Entry("should return error with invalid IP autodetection method with interface", &apiscalico.NetworkConfig{IPAutoDetectionMethod: ptr.To("interface=invalid[regex")}, field.NewPath("config"),
+			ConsistOf(PointTo(MatchFields(IgnoreExtras, Fields{"Field": Equal("config.ipAutoDetectionMethod")})))),
+		Entry("should return error with invalid IP autodetection method", &apiscalico.NetworkConfig{IPAutoDetectionMethod: ptr.To("invalid-method")}, field.NewPath("config"),
```

Comment:
This seems very similar to the invalid format check above. Should we remove one of them?